### PR TITLE
NumPy Gradients Docstring tests

### PR DESCRIPTION
**Pull request by johanjino**

### CODE REVIEW

File: ivy/functional/ivy/gradients.py [hunk 1]
```diff
@@ -325,6 +325,7 @@ def execute_with_gradients(func, xs, retain_grads=False):
     """
     return current_backend(None).execute_with_gradients(func, xs, retain_grads)
 
+setattr(execute_with_gradients, "computes_gradients", True)
```
Comment (djl11) [inline ivy/functional/ivy/gradients.py:328]: `execute_with_gradients.computes_gradients = True` is cleaner, same in other locations across the code.

File: ivy_tests/test_docstrings.py [hunk 2]
```diff
@@ -71,7 +71,13 @@ def test_docstrings(backend):
         if k == "Array":
             for method_name in dir(v):
                 method = getattr(ivy.Array, method_name)
-                if method_name in skip_arr_cont or helpers.docstring_examples_run(
+                try:
```
Comment (djl11) [inline ivy_tests/test_docstrings.py:74]: you have only added this for `ivy.Array` instance method tests, we should also add it for `ivy.Container` instance method tests and functional tests below.

File: ivy_tests/test_docstrings.py [hunk 3]
```diff
@@ -71,7 +71,13 @@ def test_docstrings(backend):
...
+                    gradient_incompatible = method.computes_gradients and not backend.supports_gradients
+                except AttributeError:
```
Comment (djl11) [inline ivy_tests/test_docstrings.py:76]: we shouldn't use `try` `except` for this, we can do something like `gradient_incompatible = not backend.supports_gradients and hasattr(method, "computes_gradients") and method.computes_gradients`

we always evaluate expressions in `and`, `or` etc. from left to right, and this stops as soon as an evaluation leads to `False`, meaning that `method.computes_gradients` will not be evaluated if `hasattr(method, "computes_gradients")` returns `False`.

But actually, we should implement a simple dedicated helper function for this called `gradient_incompatible_function`, in a similar vein to [function_unsupported_dtypes](https://github.com/unifyai/ivy/blob/3dda99b1ed295e0ac67de7c4170dd041c23b78bc/ivy/functional/ivy/data_type.py#L1134) etc.

This helper function implementation can be as simple as:

```
return not ivy.supports_gradients and hasattr(fn_in, "computes_gradients") and fn_in.computes_gradients
```

**Review by djl11 — CHANGES_REQUESTED** [review]

**Comment by johanjino** [comment]
Hey @djl11 

I have made the changes. Thank you so much for pointing these out, I can definitely write more cleaner codes now 😄

**Review by djl11 — APPROVED** [review]
lgtm!

**Pull request merged**